editor/src/clj/editor/editor_extensions/graph.clj

```diff
@@ -388,68 +410,103 @@
+  (-> attachment
+      (util/provide-defaults
+        "id" (rt/->lua (id/gen "emitter" (g/node-value parent-node-id :ids evaluation-context)))
+        "mode" default-emitter-mode
+        "space" default-emitter-space
+        "type" default-emitter-type
+        "max_particle_count" default-emitter-max-particle-count
+        "material" default-emitter-material
+        "tile_source" default-emitter-tile-source
+        "animation" default-emitter-animation
+
+        "emitter_key_particle_red" default-emitter-curve-1
+        "emitter_key_particle_green" default-emitter-curve-1
+        "emitter_key_particle_blue" default-emitter-curve-1
+        "emitter_key_particle_alpha" default-emitter-curve-1
+
+        "emitter_key_particle_life_time" default-emitter-curve-1
+        "emitter_key_particle_size" default-emitter-curve-1
+        "emitter_key_particle_speed" default-emitter-curve-1
+        "emitter_key_spawn_rate" default-emitter-curve-10
+
+        "particle_key_red" default-emitter-curve-1
+        "particle_key_green" default-emitter-curve-1
+        "particle_key_blue" default-emitter-curve-1
+        "particle_key_alpha" default-emitter-curve-1
+
+        "particle_key_scale" default-emitter-curve-1)
+      (attachment->set-tx-steps child-node-id rt project evaluation-context)))
 
 (def ^:private default-modifier-type (rt/->lua :modifier-type-acceleration))
-(defmethod init-attachment :editor.particlefx/ModifierNode [_ attachment _parent-node-id _project _evaluation-context]
-  (util/provide-defaults
-    attachment
-    "type" default-modifier-type))
-
-(defn- init-txs [evaluation-context rt project parent-node-id child-node-type child-node-id attachment]
-  (mapcat
-    (fn [[property lua-value]]
-      (if-let [setter (ext-lua-value-setter child-node-id property rt project evaluation-context)]
-        (setter lua-value)
-        (throw (LuaError. (format "Can't set property \"%s\" of %s"
-                                  property
-                                  (name (node-id->type-keyword child-node-id evaluation-context)))))))
-    (init-attachment child-node-type attachment parent-node-id project evaluation-context)))
+(defmethod init-attachment :editor.particlefx/ModifierNode [evaluation-context rt project _ _ child-node-id attachment]
+  (-> attachment
+      (util/provide-defaults
+        "type" default-modifier-type)
+      (attachment->set-tx-steps child-node-id rt project evaluation-context)))
+
+(def ^:private default-sphere-diameter (rt/->lua 20.0))
+(defmethod init-attachment :editor.collision-object/SphereShape [evaluation-context rt project _ _ child-node-id attachment]
+  (concat
+    (g/set-property child-node-id :shape-type :type-sphere)
+    (-> attachment
+        (util/provide-defaults
+          "diameter" default-sphere-diameter)
+        (attachment->set-tx-steps child-node-id rt project evaluation-context))))
```

Comment:
Refactoring of `init-attachment` was needed to support initializing the collision shapes: we set the `shape-type` property without exposing it to editor scripts, since it shouldn't be set after node construction.